README.md

```diff
@@ -6,12 +6,120 @@
+ * [API](#API)
+ * [Examples](#Examples)
+ * [Contributing](#Contributing)
+
+## Installation
+```
+ember install ember-frost-list
+```
 
 ## API
 
-Coming soon
+## Basic usage
+| parameters type | Attribute | Type | Value | Description |
+|---------| --------- | ---- | ----- | ----------- |
+| `Positional param` | `N/A` | `string` | | path for user provided list-item component which describes each row/item in the list|
+| `Attribute` | `records` | `array` | | data model used for rendering list |
+| `Attribute` | `selections` | `array` | | array of currently selected items in list |
+| `Attribute` | `defaultHeight` | `number` | | default height for each list item, set to 45px if not provided |
+| `Attribute` | `onSelect` | `<action-name>` | | triggers associated user action when item in list gets selected |
+
+### Infinite scroll
+| parameters type | Attribute | Type | Description |
+|---------| --------- | ---- | ----------- |
+| `Attribute` | `loadPrevious` | `<action-name>` | triggers associated user action when scroll reaches upper boundary |
+| `Attribute` | `loadNext` | `<action-name>` | triggers associated user action when scroll reaches lower boundary |
+
+## Examples
+### Basic list
+In order to follow the data down action up patten, `ember-frost-list` holds no state or information of current selected list items. When an item selection event emits, The user provided actions will be called and attrs will be provided to consumer as first param of the function with all useful information that describe the current select event
+
+In template
+```handlebars
+{{frost-list
+  componentPath
+  records=listItems
+  selections=selectedItems
+  onSelect=(action 'selected')
+}}
+```
+In controller
+```javascript
+componentPath: 'user-list-item',
+listItems: Ember.computed.alias('model'),
+selectedItems: [],
+actions: {
+  selected (attrs) {
+    if (attrs.isSelected) {
+      if (attrs.isShiftSelect) {
+        _.each(attrs.record, (record) => {
+          this.get('selectedItems').addObject(record)
+        })
+      } else {
+        if ((!attrs.isTargetSelectionIndicator && !attrs.isCtrlSelect)) this.set('selectedItems', [])
+        this.get('selectedItems').addObject(attrs.record)
+      }
+    } else {
+      this.get('selectedItems').removeObject(attrs.record)
+    }
+  }
+}
+```
+
+### Infinite scroll
+Implementing infinite scroll is straightforward. When scroll bar hits either upper boundary or lower boundary, loadPrevious/loadNext event will be emitted and consumer actions/functions will be called if provided.
+
+In template
+```handlebars
+{{frost-list
+  componentPath
+  records=listItems
+  selections=selectedItems
+  loadPrevious=(action 'loadPrevious')
+  loadNext=(action 'loadNext')
+  onSelect=(action 'selected')
+}}
+```
+In controller
+```javascript
+componentPath: 'user-list-item',
+listItems: Ember.computed.alias('model'),
+selectedItems: [],
+actions: {
+  _loadNext () {
+    // data retrieval/backend API call goes here
```

Comment:
I think we should show an example of working with Ember data here - we did some extra work to arrange the records in the correct order right?